edgelet/edgelet-docker/src/runtime.rs

```diff
@@ -829,6 +777,66 @@ where
     )
 }
 
+fn authenticate<MR>(
+    runtime: &MR,
+    req: &Request<Body>,
+) -> Box<dyn Future<Item = AuthId, Error = Error> + Send>
+where
+    MR: ModuleRuntime<Error = Error>,
+    <MR as ModuleRuntime>::ListFuture: 'static,
+    MR::Module: DockerModuleTop<Error = Error>,
+{
+    let pid = req
+        .extensions()
+        .get::<Pid>()
+        .cloned()
+        .unwrap_or_else(|| Pid::None);
+
+    Box::new(match pid {
+        Pid::None => Either::A(future::ok(AuthId::None)),
+        Pid::Any => Either::A(future::ok(AuthId::Any)),
+        Pid::Value(pid) => {
+            Either::B(
+                runtime
+                    .list()
+                    .into_stream()
+                    .map(|list| {
+                        stream::futures_unordered(list.into_iter().map(|module| module.top()))
+                    })
+                    .flatten()
+                    .then(Ok::<_, Error>) // wrap error into a result to ignore Module::NotFound error
```

Comment:
Do we really want to ignore a _module not found_ error? Doesn't this mean that a module is trying to authenticate the doesn't exist to docker? Seems like an exceptional situation.

Reply:
It can be several errors, that for authentication purposes we will just ignore and return `Auth::None` if someone made a call for a module that is no longer active